### DISCUSSION TITLE
Snipe-IT is not reachable after upgrade

### DISCUSSION THREAD

**Discussion by emammadov** [snipe/snipe-it#15626] [General]
Hello,
We are using Snipe-IT v6.4.2 on Ubuntu 22.04. PHP version is  8.1.23. I have updated Snipe_IT, the update process finished successfully. After the update, I can not reach the webpage of Snipe-IT. I have found the following message in /var/log/apache2/error.log. Could you please help me to solve this problem? Thanks.

`[Mon Oct 07 16:07:50.468602 2024] [php:error] [pid 3651] [client 10.200.190.109:52041] PHP Fatal error:  Uncaught ReflectionException: Class "view" does not exist in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php:877\nStack trace:\n#0 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(877): ReflectionClass->__construct()\n#1 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(1419): Illuminate\\Foundation\\Application->make()\n#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(198): Illuminate\\Container\\Container->offsetGet()\n#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(166): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()\n#8 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(255): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()\n#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/RegisterErrorViewPaths.php(18): Illuminate\\Support\\Facades\\Facade::__callStatic()\n#10 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(598): Illuminate\\Foundation\\Exceptions\\RegisterErrorViewPaths->__invoke()\n#11 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(579): Illuminate\\Foundation\\Exceptions\\Handler->registerErrorViewPaths()\n#12 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(490): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#13 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(356): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#14 /var/www/snipeit/app/Exceptions/Handler.php(133): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#15 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): App\\Exceptions\\Handler->render()\n#16 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(173): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#17 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(209): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#18 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleShutdown()\n#19 {main}\n\nNext Illuminate\\Contracts\\Container\\BindingResolutionException: Target class [view] does not exist. in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php:879\nStack trace:\n#0 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#1 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(1419): Illuminate\\Foundation\\Application->make()\n#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(198): Illuminate\\Container\\Container->offsetGet()\n#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(166): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()\n#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(255): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()\n#8 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/RegisterErrorViewPaths.php(18): Illuminate\\Support\\Facades\\Facade::__callStatic()\n#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(598): Illuminate\\Foundation\\Exceptions\\RegisterErrorViewPaths->__invoke()\n#10 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(579): Illuminate\\Foundation\\Exceptions\\Handler->registerErrorViewPaths()\n#11 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(490): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#12 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(356): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#13 /var/www/snipeit/app/Exceptions/Handler.php(133): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#14 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(196): App\\Exceptions\\Handler->render()\n#15 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(173): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#16 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(209): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#17 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleShutdown()\n#18 {main}\n  thrown in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 879
`

**Comment by snipe** [comment]
Can you show me the output of the `upgrade.php` script?

**Comment by emammadov** [comment]
Thank you for your reply. I am attaching the file of output for your review.

I have found these errors in the output, but just in case please review the whole output.

`--------------------------------------------------------
STEP 7: Cleaning up old cached files:
--------------------------------------------------------
   Error 

  Call to undefined method Illuminate\View\Compilers\BladeCompiler::anonymousComponentPath()

  at vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:261
    257▕         if (! $instance) {
    258▕             throw new RuntimeException('A facade root has not been set.');
    259▕         }
    260▕ 
  ➜ 261▕         return $instance->$method(...$args);
    262▕     }
    263▕ }
    264▕

  1   app/Providers/BladeServiceProvider.php:23
      Illuminate\Support\Facades\Facade::__callStatic()

      [2m+7 vendor frames [22m
  9   [internal]:0
      Illuminate\Foundation\Application::Illuminate\Foundation\{closure}()

--------------------------------------------------------
STEP 9: Migrating database:
--------------------------------------------------------


In ProviderRepository.php line 208:
                                                             
  Class "Facade\Ignition\IgnitionServiceProvider" not found  
                                                             


--------------------------------------------------------
STEP 10: Checking for OAuth keys:
--------------------------------------------------------

√ OAuth keys detected. Skipping passport install.

--------------------------------------------------------
STEP 11: Taking application out of maintenance mode:
--------------------------------------------------------

-- 
In ProviderRepository.php line 208:
                                                             
  Class "Facade\Ignition\IgnitionServiceProvider" not found  
`

**Comment by snipe** [comment]
Try running `composer --self-update` to update composer too the latest, then run `git pull master` and then try the upgrader again.

**Comment by emammadov** [comment]
```
snipeitapp@snipeit:/var/www/snipeit$ composer --self-update


  [Symfony\Component\Console\Exception\RuntimeException]
  The "--self-update" option does not exist.


list [--xml] [--raw] [--format FORMAT] [--] [<namespace>]
```

```
snipeitapp@snipeit:/var/www/snipeit$ sudo composer self-update 
[sudo] password for snipeitapp: 
```
I don't have a password for the user snipeitapp.

**Comment by emammadov** [comment]
I have removed composer which was installed globally, and installed it under the user snipeitapp and then upgraded Snipe-IT, and upgrade was successfully.

**Comment by snipe** [comment]
Great, glad you're sorted!
